packages/compiler/src/jit_compiler_facade.ts

```diff
@@ -482,11 +482,16 @@ function convertR3DependencyMetadataArray(facades: R3DependencyMetadataFacade[]|
 function convertR3DeclareDependencyMetadata(
     {token, attribute, host = false, optional = false, self = false, skipSelf = false}:
         R3DeclareDependencyMetadataFacade): R3DependencyMetadata {
-  const dep: R3DependencyMetadata =
-      {token: token === null ? null : new WrappedNodeExpr(token), host, optional, self, skipSelf};
-  if (attribute !== undefined) {
-    dep.attribute = new WrappedNodeExpr(attribute);
-  }
+  const wrappedAttribute = attribute === undefined ? undefined : new WrappedNodeExpr(attribute);
+  const wrappedToken = wrappedAttribute || token === null ? null : new WrappedNodeExpr(token);
```

Comment:
My brain has trouble deriving the precedence here... would this be

```suggestion
  const wrappedToken = wrappedAttribute ?? (token === null ? null : new WrappedNodeExpr(token));
```

?

Reply:
Other refactoring has made this suggestion redundant.